src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1639,13 +1661,38 @@ private async Task ClearMarkers(ScriptFile scriptFile, EventContext eventContext
                 });
         }
 
+        // Generate a unique id that is used as a key to look up the associated code action (code fix) when 
+        // we receive and process the textDocument/codeAction message.
+        private static string GetUniqueIdFromDiagnostic(Diagnostic diagnostic)
+        {
+            Position start = diagnostic.Range.Start;
+            Position end = diagnostic.Range.End;
+
+            var sb = new StringBuilder(256);
+            sb.Append(diagnostic.Source ?? "?");
+            sb.Append("_");
+            sb.Append(diagnostic.Code ?? "?");
+            sb.Append("_");
+            sb.Append((diagnostic.Severity != null) ? diagnostic.Severity.ToString() : "?");
+            sb.Append("_");
+            sb.Append(start.Line);
+            sb.Append(":");
+            sb.Append(start.Character);
+            sb.Append("-");
+            sb.Append(end.Line);
+            sb.Append(":");
+            sb.Append(end.Character);
```

Comment:
small nit... it might be useful to do:

```csharp
StringBuilder sb = new StringBuilder(256)
    .Append(diagnostic.Source ?? "?")
    .Append("_")
    .Append(diagnostic.Code ?? "?")
    .Append("_")
    .Append((diagnostic.Severity != null) ? diagnostic.Severity.ToString() : "?")
    .Append("_")
    .Append(start.Line)
    .Append(":")
    .Append(start.Character)
    .Append("-")
    .Append(end.Line)
    .Append(":")
    .Append(end.Character)

return sb.ToString();
```

so that we can throw a breakpoint on the return line and see what the contents of sb are.

Reply:
Heh, I was just going to suggest this very thing.  :-)  Will fix.

Reply:
Honestly my biggest desire is for a function-return-variable in debug view.